assets/scripts/build_gallery.py

```diff
@@ -30,7 +114,21 @@ def build_gallery(include_core=True, include_contrib_experimental=True):
         expectations_module = importlib.import_module(
             "expectations", "great_expectations_experimental"
         )
+        requirements_dict = {}
+        for root, dirs, files in os.walk(contrib_experimental_dir):
+            for file in files:
+                if file.endswith(".py") and not file == "__init__.py":
+                    logger.debug(f"Getting requirements for module {file}")
+                    requirements_dict[file[:-3]] = get_contrib_requirements(
+                        os.path.join(root, file)
+                    )
+
         for expectation_module in expectations_module.__all__:
+            if expectation_module in requirements_dict:
```

Comment:
Nitpick, the install step for expectation and metric dependencies could be moved into a separate method.

Reply:
I realized that we don't have run_diagnostics for metrics, so clarified that and added a warning if one is discovered.